napari/components/dims.py

```diff
@@ -353,28 +300,30 @@ def set_point(self, axis: int, value: Union[int, float]):
         ----------
         axis : int
             Dimension index.
-        value : int or float
+        point : int or float
             Value of the point.
         """
         axis = self._assert_axis_in_bounds(axis)
-        if self.point[axis] != value:
-            self._point[axis] = value
-            self.events.axis(axis=axis, value=value)
+        (min_val, max_val, step_size) = self._range[axis]
+        raw_step = (value - min_val) / step_size
+        self.set_step(axis, raw_step)
 
-    def set_interval(self, axis: int, interval: Sequence[Union[int, float]]):
-        """Sets the interval used for cropping and projecting this dimension.
+    def set_step(self, axis: int, value: int):
```

Comment:
I don't find this self-explanatory at all... From the name, I would expect this to be equivalent to `self.set_range(axis, *self.range[:2], step)`, ie setting the step size. Maybe `self.set_step_number`? Or `self.set_point(axis, pt, coordinates='data')`? (as opposed to `'world'`)

Reply:
`set_current_step`?

Reply:
I also don't like that set_point now depends on this. The vision for set_point is that it should be a continuous value, and future applications might include for example setting the point and then using that point to pivot oblique cut planes. For that, you need the point *before* rounding. So it seems to me that this should be changed so that:

- `set_point` is the "canonical" API and keeps all the info (including fractional coordinates)
- the `step` property (maybe `current_step`?) grabs the point attribute and rounds it appropriately using `range`.
- `set_step` uses `range` to call `set_point`.

What do you think @sofroniewn? Is this something you tried before and it failed?

Reply:
> What do you think @sofroniewn? Is this something you tried before and it failed?

you are correct, this is where i started, for the reasoning you provided, but it created all sorts of weird behavior due to the round trip nature of setting the slider value and the slider value then corresponding to a point value. Some of that may be symptomatic of deeper issues that could get resolved, but it was much more awkward from an implementation stand point to do it that way.

Reply:
Hmmm. Ok. I guess we can punt on this, but it makes me uncomfortable! All this might fall out nicely in #1353 anyway. Perhaps add a comment in both docstrings (set_step/set_point) that make it clear how each is implemented in terms of the other, vs the ideal case?

Reply:
I've gone with `current_step` and added comments. I am leaving the `point` vs `current_step` as they are now, but agreed we might want to revisit. As noted above I wanted to do it that way originally